<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <id>https://azu.github.io/github-advisory-database-rss/pip.rss</id>
    <title>Security Advisory for Python packages hosted at PyPI.org</title>
    <updated>2024-11-18T23:01:30.519Z</updated>
    <generator>github-advisory-database-rss</generator>
    <link rel="alternate" href="https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"/>
    <subtitle>Security Advisory for Python packages hosted at PyPI.org on GitHub</subtitle>
    <rights>github-advisory-database-rss</rights>
    <category term="CRITICAL"/>
    <category term="HIGH"/>
    <category term="MODERATE"/>
    <category term="LOW"/>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton vulnerable to arbitrary command execution]]></title>
        <id>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</id>
        <link href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx"/>
        <updated>2024-11-18T22:55:29.000Z</updated>
        <content type="html"><![CDATA[<p>The <code>safe_eval</code> function in trytond in Tryton before 2.4.15, 2.6.x before 2.6.14, 2.8.x before 2.8.11, 3.0.x before 3.0.7, and 3.2.x before 3.2.3 allows remote authenticated users to execute arbitrary commands via shell metacharacters in (1) the <code>collection.domain</code> in the webdav module or (2) the formula field in the <code>price_list</code> module.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-6633">https://nvd.nist.gov/vuln/detail/CVE-2014-6633</a></li>
<li><a href="https://bugs.tryton.org/issue4155">https://bugs.tryton.org/issue4155</a></li>
<li><a href="http://www.tryton.org/posts/security-release-for-issue4155.html">http://www.tryton.org/posts/security-release-for-issue4155.html</a></li>
<li><a href="https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06">https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06</a></li>
<li><a href="https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a">https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx">https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:21:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton vulnerable to arbitrary command execution]]></title>
        <id>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</id>
        <link href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx"/>
        <updated>2024-11-18T22:55:29.000Z</updated>
        <content type="html"><![CDATA[<p>The <code>safe_eval</code> function in trytond in Tryton before 2.4.15, 2.6.x before 2.6.14, 2.8.x before 2.8.11, 3.0.x before 3.0.7, and 3.2.x before 3.2.3 allows remote authenticated users to execute arbitrary commands via shell metacharacters in (1) the <code>collection.domain</code> in the webdav module or (2) the formula field in the <code>price_list</code> module.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-6633">https://nvd.nist.gov/vuln/detail/CVE-2014-6633</a></li>
<li><a href="https://bugs.tryton.org/issue4155">https://bugs.tryton.org/issue4155</a></li>
<li><a href="http://www.tryton.org/posts/security-release-for-issue4155.html">http://www.tryton.org/posts/security-release-for-issue4155.html</a></li>
<li><a href="https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06">https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06</a></li>
<li><a href="https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a">https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx">https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:21:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton vulnerable to arbitrary command execution]]></title>
        <id>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</id>
        <link href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx"/>
        <updated>2024-11-18T22:55:29.000Z</updated>
        <content type="html"><![CDATA[<p>The <code>safe_eval</code> function in trytond in Tryton before 2.4.15, 2.6.x before 2.6.14, 2.8.x before 2.8.11, 3.0.x before 3.0.7, and 3.2.x before 3.2.3 allows remote authenticated users to execute arbitrary commands via shell metacharacters in (1) the <code>collection.domain</code> in the webdav module or (2) the formula field in the <code>price_list</code> module.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-6633">https://nvd.nist.gov/vuln/detail/CVE-2014-6633</a></li>
<li><a href="https://bugs.tryton.org/issue4155">https://bugs.tryton.org/issue4155</a></li>
<li><a href="http://www.tryton.org/posts/security-release-for-issue4155.html">http://www.tryton.org/posts/security-release-for-issue4155.html</a></li>
<li><a href="https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06">https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06</a></li>
<li><a href="https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a">https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx">https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:21:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton vulnerable to arbitrary command execution]]></title>
        <id>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</id>
        <link href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx"/>
        <updated>2024-11-18T22:55:29.000Z</updated>
        <content type="html"><![CDATA[<p>The <code>safe_eval</code> function in trytond in Tryton before 2.4.15, 2.6.x before 2.6.14, 2.8.x before 2.8.11, 3.0.x before 3.0.7, and 3.2.x before 3.2.3 allows remote authenticated users to execute arbitrary commands via shell metacharacters in (1) the <code>collection.domain</code> in the webdav module or (2) the formula field in the <code>price_list</code> module.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-6633">https://nvd.nist.gov/vuln/detail/CVE-2014-6633</a></li>
<li><a href="https://bugs.tryton.org/issue4155">https://bugs.tryton.org/issue4155</a></li>
<li><a href="http://www.tryton.org/posts/security-release-for-issue4155.html">http://www.tryton.org/posts/security-release-for-issue4155.html</a></li>
<li><a href="https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06">https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06</a></li>
<li><a href="https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a">https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx">https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:21:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton vulnerable to arbitrary command execution]]></title>
        <id>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</id>
        <link href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx"/>
        <updated>2024-11-18T22:55:29.000Z</updated>
        <content type="html"><![CDATA[<p>The <code>safe_eval</code> function in trytond in Tryton before 2.4.15, 2.6.x before 2.6.14, 2.8.x before 2.8.11, 3.0.x before 3.0.7, and 3.2.x before 3.2.3 allows remote authenticated users to execute arbitrary commands via shell metacharacters in (1) the <code>collection.domain</code> in the webdav module or (2) the formula field in the <code>price_list</code> module.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-6633">https://nvd.nist.gov/vuln/detail/CVE-2014-6633</a></li>
<li><a href="https://bugs.tryton.org/issue4155">https://bugs.tryton.org/issue4155</a></li>
<li><a href="http://www.tryton.org/posts/security-release-for-issue4155.html">http://www.tryton.org/posts/security-release-for-issue4155.html</a></li>
<li><a href="https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06">https://github.com/tryton/trytond/commit/19fc2a01357b7638041953326e404f51d96fad06</a></li>
<li><a href="https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a">https://github.com/tryton/trytond/commit/3e4c2b7e8c7b3358597a0d484fa98f45483ee92a</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2018-59.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m9jj-5qvj-5fhx">https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m9jj-5qvj-5fhx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:21:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[trytond] Tryton Information Disclosure Vulnerability]]></title>
        <id>https://github.com/advisories/GHSA-7cwg-2575-3546</id>
        <link href="https://github.com/advisories/GHSA-7cwg-2575-3546"/>
        <updated>2024-11-18T22:54:24.000Z</updated>
        <content type="html"><![CDATA[<p>file_open in Tryton 3.x and 4.x through 4.2.2 allows remote authenticated users with certain permissions to read arbitrary files via a "same root name but with a suffix" attack. NOTE: This vulnerability exists because of an incomplete fix for CVE-2016-1242.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2017-0360">https://nvd.nist.gov/vuln/detail/CVE-2017-0360</a></li>
<li><a href="https://lists.debian.org/debian-security-announce/2017/msg00084.html">https://lists.debian.org/debian-security-announce/2017/msg00084.html</a></li>
<li><a href="http://hg.tryton.org/trytond?cmd=changeset;node=472510fdc6f8">http://hg.tryton.org/trytond?cmd=changeset;node=472510fdc6f8</a></li>
<li><a href="http://www.debian.org/security/2017/dsa-3826">http://www.debian.org/security/2017/dsa-3826</a></li>
<li><a href="https://github.com/tryton/trytond/commit/30e978593733385db3144f8c583eeb4679575cf0">https://github.com/tryton/trytond/commit/30e978593733385db3144f8c583eeb4679575cf0</a></li>
<li><a href="https://github.com/tryton/trytond/commit/a67a7f03c30277515f530cad5950056171ed5bd1">https://github.com/tryton/trytond/commit/a67a7f03c30277515f530cad5950056171ed5bd1</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2017-97.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/trytond/PYSEC-2017-97.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-7cwg-2575-3546">https://github.com/advisories/GHSA-7cwg-2575-3546</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-7cwg-2575-3546</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2022-05-13T01:39:57.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[twisted] Twisted SSH client and server deny of service during SSH handshake.]]></title>
        <id>https://github.com/advisories/GHSA-rv6r-3f5q-9rgx</id>
        <link href="https://github.com/advisories/GHSA-rv6r-3f5q-9rgx"/>
        <updated>2024-11-18T22:50:33.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>The Twisted SSH client and server implementation naively accepted an infinite amount of data for the  peer's SSH version identifier.</p>
<p>A malicious peer can trivially craft a request that uses all available memory and crash the server, resulting in denial of service. The attack is as simple as <code>nc -rv localhost 22 &lt; /dev/zero</code>.</p>
<h3 id="patches">Patches</h3>
<p>The issue was fix in GitHub commit <a href="https://github.com/twisted/twisted/commit/98387b39e9f0b21462f6abc7a1325dc370fcdeb1">https://github.com/twisted/twisted/commit/98387b39e9f0b21462f6abc7a1325dc370fcdeb1</a></p>
<p>A fix is available in Twisted 22.2.0.</p>
<h3 id="workarounds">Workarounds</h3>
<ul>
<li>Limit access to the SSH server only to trusted source IP addresses.</li>
<li>Connect over SSH only to trusted destination IP addresses.</li>
</ul>
<h3 id="references">References</h3>
<p>Reported at <a href="https://twistedmatrix.com/trac/ticket/10284">https://twistedmatrix.com/trac/ticket/10284</a>
Discussions at <a href="https://github.com/twisted/twisted/security/advisories/GHSA-rv6r-3f5q-9rgx">https://github.com/twisted/twisted/security/advisories/GHSA-rv6r-3f5q-9rgx</a></p>
<h3 id="for-more-information">For more information</h3>
<p>Found by vin01</p>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/twisted/twisted/security/advisories/GHSA-rv6r-3f5q-9rgx">https://github.com/twisted/twisted/security/advisories/GHSA-rv6r-3f5q-9rgx</a></li>
<li><a href="https://github.com/twisted/twisted/commit/98387b39e9f0b21462f6abc7a1325dc370fcdeb1">https://github.com/twisted/twisted/commit/98387b39e9f0b21462f6abc7a1325dc370fcdeb1</a></li>
<li><a href="https://twistedmatrix.com/trac/ticket/10284">https://twistedmatrix.com/trac/ticket/10284</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2022-21716">https://nvd.nist.gov/vuln/detail/CVE-2022-21716</a></li>
<li><a href="https://github.com/twisted/twisted/releases/tag/twisted-22.2.0">https://github.com/twisted/twisted/releases/tag/twisted-22.2.0</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2022/03/msg00009.html">https://lists.debian.org/debian-lts-announce/2022/03/msg00009.html</a></li>
<li><a href="https://www.oracle.com/security-alerts/cpuapr2022.html">https://www.oracle.com/security-alerts/cpuapr2022.html</a></li>
<li><a href="https://security.gentoo.org/glsa/202301-02">https://security.gentoo.org/glsa/202301-02</a></li>
<li><a href="https://github.com/twisted/twisted/commit/89c395ee794e85a9657b112c4351417850330ef9">https://github.com/twisted/twisted/commit/89c395ee794e85a9657b112c4351417850330ef9</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2022-160.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2022-160.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7U6KYDTOLPICAVSR34G2WRYLFBD2YW5K">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7U6KYDTOLPICAVSR34G2WRYLFBD2YW5K</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GLKHA6WREIVAMBQD7KKWYHPHGGNKMAG6">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GLKHA6WREIVAMBQD7KKWYHPHGGNKMAG6</a></li>
<li><a href="https://github.com/advisories/GHSA-rv6r-3f5q-9rgx">https://github.com/advisories/GHSA-rv6r-3f5q-9rgx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-rv6r-3f5q-9rgx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-03-03T19:02:08.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[Twisted] Cookie and header exposure in twisted]]></title>
        <id>https://github.com/advisories/GHSA-92x2-jw7w-xvvx</id>
        <link href="https://github.com/advisories/GHSA-92x2-jw7w-xvvx"/>
        <updated>2024-11-18T22:48:33.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>Cookie and Authorization headers are leaked when following cross-origin redirects in <code>twited.web.client.RedirectAgent</code> and <code>twisted.web.client.BrowserLikeRedirectAgent</code>.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/twisted/twisted/security/advisories/GHSA-92x2-jw7w-xvvx">https://github.com/twisted/twisted/security/advisories/GHSA-92x2-jw7w-xvvx</a></li>
<li><a href="https://github.com/twisted/twisted/commit/af8fe78542a6f2bf2235ccee8158d9c88d31e8e2">https://github.com/twisted/twisted/commit/af8fe78542a6f2bf2235ccee8158d9c88d31e8e2</a></li>
<li><a href="https://github.com/twisted/twisted/releases/tag/twisted-22.1.0">https://github.com/twisted/twisted/releases/tag/twisted-22.1.0</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2022-21712">https://nvd.nist.gov/vuln/detail/CVE-2022-21712</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2022/02/msg00021.html">https://lists.debian.org/debian-lts-announce/2022/02/msg00021.html</a></li>
<li><a href="https://security.gentoo.org/glsa/202301-02">https://security.gentoo.org/glsa/202301-02</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2022-27.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2022-27.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7U6KYDTOLPICAVSR34G2WRYLFBD2YW5K">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7U6KYDTOLPICAVSR34G2WRYLFBD2YW5K</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GLKHA6WREIVAMBQD7KKWYHPHGGNKMAG6">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GLKHA6WREIVAMBQD7KKWYHPHGGNKMAG6</a></li>
<li><a href="https://pypi.org/project/Twisted">https://pypi.org/project/Twisted</a></li>
<li><a href="https://github.com/advisories/GHSA-92x2-jw7w-xvvx">https://github.com/advisories/GHSA-92x2-jw7w-xvvx</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-92x2-jw7w-xvvx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-02-07T22:36:00.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tuf] Client metadata path-traversal]]></title>
        <id>https://github.com/advisories/GHSA-wjw6-2cqr-j4qr</id>
        <link href="https://github.com/advisories/GHSA-wjw6-2cqr-j4qr"/>
        <updated>2024-11-18T22:45:47.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>In both clients (<code>tuf/client</code> and <code>tuf/ngclient</code>), there is a path traversal vulnerability that in the worst case can overwrite files ending in <code>.json</code> anywhere on the client system on a call to <code>get_one_valid_targetinfo()</code>. It occurs because the rolename is used to form the filename, and may contain path traversal characters (ie <code>../../name.json</code>).</p>
<p>The impact is mitigated by a few facts:</p>
<ul>
<li>It only affects implementations that allow arbitrary rolename selection for delegated targets metadata</li>
<li>The attack requires the ability to A) insert new metadata for the path-traversing role and B) get the role delegated by an existing targets metadata</li>
<li>The written file content is heavily restricted since it needs to be a valid, signed targets file. The file extension is always .json.</li>
</ul>
<h3 id="patches">Patches</h3>
<p>A fix is available in version 0.19 or newer.</p>
<h3 id="workarounds">Workarounds</h3>
<p>None that do not require code changes. Clients can restrict the allowed character set for rolenames, or they can store metadata in files named in a way that is not vulnerable: neither of these approaches is possible without modifying python-tuf.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/theupdateframework/python-tuf/issues/1527">The issue where this was discovered</a></li>
<li><a href="https://github.com/jku/path-traversal-poc">Proof of Concept demonstrating the flaw</a></li>
</ul>
<h3 id="for-more-information">For more information</h3>
<p>If you have any questions or comments about this advisory:</p>
<ul>
<li>Open an issue in <a href="https://github.com/theupdateframework/python-tuf/issues">python-tuf</a></li>
<li>Contact the maintainers by email or Slack</li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/theupdateframework/python-tuf/security/advisories/GHSA-wjw6-2cqr-j4qr">https://github.com/theupdateframework/python-tuf/security/advisories/GHSA-wjw6-2cqr-j4qr</a></li>
<li><a href="https://github.com/theupdateframework/python-tuf/issues/1527">https://github.com/theupdateframework/python-tuf/issues/1527</a></li>
<li><a href="https://github.com/theupdateframework/python-tuf/commit/4ad7ae48fda594b640139c3b7eae21ed5155a102">https://github.com/theupdateframework/python-tuf/commit/4ad7ae48fda594b640139c3b7eae21ed5155a102</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2021-41131">https://nvd.nist.gov/vuln/detail/CVE-2021-41131</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2021-376.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2021-376.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-wjw6-2cqr-j4qr">https://github.com/advisories/GHSA-wjw6-2cqr-j4qr</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-wjw6-2cqr-j4qr</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2021-10-19T20:14:36.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tuf] Invalid root may become trusted root in The Update Framework (TUF)]]></title>
        <id>https://github.com/advisories/GHSA-f8mr-jv2c-v8mg</id>
        <link href="https://github.com/advisories/GHSA-f8mr-jv2c-v8mg"/>
        <updated>2024-11-18T22:41:33.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>The Python TUF reference implementation <code>tuf&lt;0.12</code> will incorrectly trust a previously downloaded root metadata file which failed verification at download time. This allows an attacker who is able to serve multiple new versions of root metadata (i.e. by a man-in-the-middle attack) culminating in a version which has not been correctly signed to control the trust chain for future updates.</p>
<p>While investigating the reported vulnerability, we discovered that the detailed client workflow was not fully implemented. Specifically, for step 1.3 the newly downloaded root metadata was not being verified with a threshold of keys specified in the new root metadata file.
This missing step of the client workflow has been implemented in <a href="https://github.com/theupdateframework/tuf/pull/1101">PR #1101</a>, which is included in <a href="https://github.com/theupdateframework/tuf/releases/tag/v0.14.0">v0.14.0</a> of tuf.</p>
<h3 id="patches">Patches</h3>
<p>A <a href="https://github.com/theupdateframework/tuf/pull/885">fix</a>, is available in version <a href="https://github.com/theupdateframework/tuf/releases/tag/v0.12.0">0.12</a> and newer.</p>
<h3 id="workarounds">Workarounds</h3>
<p>No workarounds are known for this issue.</p>
<h3 id="references">References</h3>
<ul>
<li>Pull request resolving the invalid root becoming trusted issue <a href="https://github.com/theupdateframework/tuf/pull/885">PR 885</a></li>
<li>Pull request implementing self verification of newly downloaded root metadata <a href="https://github.com/theupdateframework/tuf/pull/1101">PR 1101</a></li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/theupdateframework/tuf/security/advisories/GHSA-f8mr-jv2c-v8mg">https://github.com/theupdateframework/tuf/security/advisories/GHSA-f8mr-jv2c-v8mg</a></li>
<li><a href="https://github.com/theupdateframework/tuf/pull/885">https://github.com/theupdateframework/tuf/pull/885</a></li>
<li><a href="https://github.com/theupdateframework/tuf/commit/3d342e648fbacdf43a13d7ba8886aaaf07334af7">https://github.com/theupdateframework/tuf/commit/3d342e648fbacdf43a13d7ba8886aaaf07334af7</a></li>
<li><a href="https://github.com/theupdateframework/tuf/releases/tag/v0.12.0">https://github.com/theupdateframework/tuf/releases/tag/v0.12.0</a></li>
<li><a href="https://pypi.org/project/tuf">https://pypi.org/project/tuf</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-15163">https://nvd.nist.gov/vuln/detail/CVE-2020-15163</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-145.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-145.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-f8mr-jv2c-v8mg">https://github.com/advisories/GHSA-f8mr-jv2c-v8mg</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-f8mr-jv2c-v8mg</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2020-09-09T17:29:27.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tuf] Incorrect threshold signature computation in TUF]]></title>
        <id>https://github.com/advisories/GHSA-pwqf-9h7j-7mv8</id>
        <link href="https://github.com/advisories/GHSA-pwqf-9h7j-7mv8"/>
        <updated>2024-11-18T22:40:37.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>Metadadata signature verification, as used in <code>tuf.client.updater</code>, counted each of multiple signatures with identical authorized keyids  separately towards the threshold. Therefore, an attacker with access to a valid signing key could create multiple valid signatures in order to meet the minimum threshold of keys before the metadata was considered valid.</p>
<p>The tuf maintainers would like to thank Erik MacLean of Analog Devices, Inc. for reporting this issue.</p>
<h3 id="patches">Patches</h3>
<p>A <a href="https://github.com/theupdateframework/tuf/pull/974">fix</a> is available in version <a href="https://github.com/theupdateframework/tuf/releases/tag/v0.12.2">0.12.2</a> or newer.</p>
<h3 id="workarounds">Workarounds</h3>
<p>No workarounds are known for this issue.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-6174">CVE-2020-6174</a></li>
<li>Pull request resolving the issue <a href="https://github.com/theupdateframework/tuf/pull/974">PR 974</a></li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/theupdateframework/tuf/security/advisories/GHSA-pwqf-9h7j-7mv8">https://github.com/theupdateframework/tuf/security/advisories/GHSA-pwqf-9h7j-7mv8</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-6174">https://nvd.nist.gov/vuln/detail/CVE-2020-6174</a></li>
<li><a href="https://github.com/theupdateframework/tuf/pull/974">https://github.com/theupdateframework/tuf/pull/974</a></li>
<li><a href="https://github.com/theupdateframework/python-tuf/commit/2977188139d065ff3356c3cb4aec60c582b57e0e">https://github.com/theupdateframework/python-tuf/commit/2977188139d065ff3356c3cb4aec60c582b57e0e</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-147.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-147.yaml</a></li>
<li><a href="https://github.com/theupdateframework/tuf/releases/tag/v0.12.2">https://github.com/theupdateframework/tuf/releases/tag/v0.12.2</a></li>
<li><a href="https://github.com/advisories/GHSA-pwqf-9h7j-7mv8">https://github.com/advisories/GHSA-pwqf-9h7j-7mv8</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-pwqf-9h7j-7mv8</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2020-08-21T16:25:26.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tuf] Client Denial of Service on TUF]]></title>
        <id>https://github.com/advisories/GHSA-2828-9vh6-9m6j</id>
        <link href="https://github.com/advisories/GHSA-2828-9vh6-9m6j"/>
        <updated>2024-11-18T22:38:25.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>An attacker who can gain file access to the repository and modify metadata files may cause a denial of service to clients by creating many invalid signatures on a metadata file. Having a large number of signatures to verify will delay the moment when the client will determine the signature is not valid. This delay may be for at least a few minutes, but possibly could be longer especially if multiple files are impacted.</p>
<p>The tuf maintainers would like to thank Erik MacLean of Analog Devices, Inc. for reporting this issue.</p>
<h3 id="patches">Patches</h3>
<p>No fix exists for this issue. </p>
<h3 id="workarounds">Workarounds</h3>
<p>No workarounds are known for this issue.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-6173">CVE-2020-6173</a></li>
<li><a href="https://github.com/theupdateframework/tuf/issues/973">Issue #973</a></li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/theupdateframework/tuf/security/advisories/GHSA-2828-9vh6-9m6j">https://github.com/theupdateframework/tuf/security/advisories/GHSA-2828-9vh6-9m6j</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-6173">https://nvd.nist.gov/vuln/detail/CVE-2020-6173</a></li>
<li><a href="https://github.com/theupdateframework/tuf/issues/973">https://github.com/theupdateframework/tuf/issues/973</a></li>
<li><a href="https://github.com/theupdateframework/tuf/commits/develop">https://github.com/theupdateframework/tuf/commits/develop</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-146.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tuf/PYSEC-2020-146.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-2828-9vh6-9m6j">https://github.com/advisories/GHSA-2828-9vh6-9m6j</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-2828-9vh6-9m6j</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2020-08-21T16:25:48.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[urllib3] Exposure of Sensitive Information to an Unauthorized Actor in urllib3]]></title>
        <id>https://github.com/advisories/GHSA-www2-v7xj-xrc6</id>
        <link href="https://github.com/advisories/GHSA-www2-v7xj-xrc6"/>
        <updated>2024-11-18T22:31:18.000Z</updated>
        <content type="html"><![CDATA[<p>urllib3 before version 1.23 does not remove the Authorization HTTP header when following a cross-origin redirect (i.e., a redirect that differs in host, port, or scheme). This can allow for credentials in the Authorization header to be exposed to unintended hosts or transmitted in cleartext.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2018-20060">https://nvd.nist.gov/vuln/detail/CVE-2018-20060</a></li>
<li><a href="https://github.com/urllib3/urllib3/issues/1316">https://github.com/urllib3/urllib3/issues/1316</a></li>
<li><a href="https://github.com/urllib3/urllib3/pull/1346">https://github.com/urllib3/urllib3/pull/1346</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:2272">https://access.redhat.com/errata/RHSA-2019:2272</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=1649153">https://bugzilla.redhat.com/show_bug.cgi?id=1649153</a></li>
<li><a href="https://github.com/urllib3/urllib3/blob/master/CHANGES.rst">https://github.com/urllib3/urllib3/blob/master/CHANGES.rst</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2021/06/msg00015.html">https://lists.debian.org/debian-lts-announce/2021/06/msg00015.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00039.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00039.html</a></li>
<li><a href="https://github.com/urllib3/urllib3/commit/560bd227b90f74417ffaedebf5f8d05a8ee4f532">https://github.com/urllib3/urllib3/commit/560bd227b90f74417ffaedebf5f8d05a8ee4f532</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2018-32.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2018-32.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5SJERZEJDSUYQP7BNBXMBHRHGY26HRZD">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5SJERZEJDSUYQP7BNBXMBHRHGY26HRZD</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BXLAXHM3Z6DUCXZ7ZXZ2EAYJXWDCZFCT">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BXLAXHM3Z6DUCXZ7ZXZ2EAYJXWDCZFCT</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XWP36YW3KSVLXDBY3QJKDYEPCIMN3VQZ">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XWP36YW3KSVLXDBY3QJKDYEPCIMN3VQZ</a></li>
<li><a href="https://usn.ubuntu.com/3990-1">https://usn.ubuntu.com/3990-1</a></li>
<li><a href="https://github.com/advisories/GHSA-www2-v7xj-xrc6">https://github.com/advisories/GHSA-www2-v7xj-xrc6</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-www2-v7xj-xrc6</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2018-12-12T15:52:07.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[Twisted] HTTP Request Smuggling in Twisted]]></title>
        <id>https://github.com/advisories/GHSA-p5xh-vx83-mxcj</id>
        <link href="https://github.com/advisories/GHSA-p5xh-vx83-mxcj"/>
        <updated>2024-11-18T22:25:00.000Z</updated>
        <content type="html"><![CDATA[<p>In Twisted Web through 20.3.0, there was an HTTP request splitting vulnerability. When presented with a content-length and a chunked encoding header, the content-length took precedence and the remainder of the request body was interpreted as a pipelined request.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-10109">https://nvd.nist.gov/vuln/detail/CVE-2020-10109</a></li>
<li><a href="https://security.gentoo.org/glsa/202007-24">https://security.gentoo.org/glsa/202007-24</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2022/02/msg00021.html">https://lists.debian.org/debian-lts-announce/2022/02/msg00021.html</a></li>
<li><a href="https://github.com/twisted/twisted/commit/4a7d22e490bb8ff836892cc99a1f54b85ccb0281">https://github.com/twisted/twisted/commit/4a7d22e490bb8ff836892cc99a1f54b85ccb0281</a></li>
<li><a href="https://github.com/twisted/twisted/blob/6ff2c40e42416c83203422ff70dfc49d2681c8e2/NEWS.rst#twisted-2030-2020-03-13">https://github.com/twisted/twisted/blob/6ff2c40e42416c83203422ff70dfc49d2681c8e2/NEWS.rst#twisted-2030-2020-03-13</a></li>
<li><a href="https://know.bishopfox.com/advisories/twisted-version-19.10.0">https://know.bishopfox.com/advisories/twisted-version-19.10.0</a></li>
<li><a href="https://github.com/advisories/GHSA-p5xh-vx83-mxcj">https://github.com/advisories/GHSA-p5xh-vx83-mxcj</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2020-260.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2020-260.yaml</a></li>
<li><a href="https://know.bishopfox.com/advisories">https://know.bishopfox.com/advisories</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6ISMZFZBWW4EV6ETJGXAYIXN3AT7GBPL">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6ISMZFZBWW4EV6ETJGXAYIXN3AT7GBPL</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YW3NIL7VXSGJND2Q4BSXM3CFTAFU6T7D">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YW3NIL7VXSGJND2Q4BSXM3CFTAFU6T7D</a></li>
<li><a href="https://usn.ubuntu.com/4308-1">https://usn.ubuntu.com/4308-1</a></li>
<li><a href="https://usn.ubuntu.com/4308-2">https://usn.ubuntu.com/4308-2</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-p5xh-vx83-mxcj</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2020-03-31T15:40:12.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[typed-ast] typed-ast Out-of-bounds Read]]></title>
        <id>https://github.com/advisories/GHSA-7xxv-wpxj-mx5v</id>
        <link href="https://github.com/advisories/GHSA-7xxv-wpxj-mx5v"/>
        <updated>2024-11-18T22:16:28.000Z</updated>
        <content type="html"><![CDATA[<p>typed_ast 1.3.0 and 1.3.1 has an ast_for_arguments out-of-bounds read. An attacker with the ability to cause a Python interpreter to parse Python source (but not necessarily execute it) may be able to crash the interpreter process. This could be a concern, for example, in a web-based service that parses (but does not execute) Python code. (This issue also affected certain Python 3.8.0-alpha prereleases.)</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-19275">https://nvd.nist.gov/vuln/detail/CVE-2019-19275</a></li>
<li><a href="https://github.com/python/cpython/commit/a4d78362397fc3bced6ea80fbc7b5f4827aec55e">https://github.com/python/cpython/commit/a4d78362397fc3bced6ea80fbc7b5f4827aec55e</a></li>
<li><a href="https://github.com/python/cpython/commit/dcfcd146f8e6fc5c2fc16a4c192a0c5f5ca8c53c">https://github.com/python/cpython/commit/dcfcd146f8e6fc5c2fc16a4c192a0c5f5ca8c53c</a></li>
<li><a href="https://github.com/python/typed_ast/commit/156afcb26c198e162504a57caddfe0acd9ed7dce">https://github.com/python/typed_ast/commit/156afcb26c198e162504a57caddfe0acd9ed7dce</a></li>
<li><a href="https://github.com/python/typed_ast/commit/dc317ac9cff859aa84eeabe03fb5004982545b3b">https://github.com/python/typed_ast/commit/dc317ac9cff859aa84eeabe03fb5004982545b3b</a></li>
<li><a href="https://bugs.python.org/issue36495">https://bugs.python.org/issue36495</a></li>
<li><a href="https://github.com/advisories/GHSA-7xxv-wpxj-mx5v">https://github.com/advisories/GHSA-7xxv-wpxj-mx5v</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/typed-ast/PYSEC-2019-131.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/typed-ast/PYSEC-2019-131.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LG5H4Q6LFVRX7SFXLBEJMNQFI4T5SCEA">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LG5H4Q6LFVRX7SFXLBEJMNQFI4T5SCEA</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-7xxv-wpxj-mx5v</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2019-12-02T18:03:09.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[twisted] Improper Certificate Validation in Twisted]]></title>
        <id>https://github.com/advisories/GHSA-65rm-h285-5cc5</id>
        <link href="https://github.com/advisories/GHSA-65rm-h285-5cc5"/>
        <updated>2024-11-18T22:12:14.000Z</updated>
        <content type="html"><![CDATA[<p>In words.protocols.jabber.xmlstream in Twisted through 19.2.1, XMPP support did not verify certificates when used with TLS, allowing an attacker to MITM connections.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-12855">https://nvd.nist.gov/vuln/detail/CVE-2019-12855</a></li>
<li><a href="https://github.com/twisted/twisted/pull/1147">https://github.com/twisted/twisted/pull/1147</a></li>
<li><a href="https://twistedmatrix.com/trac/ticket/9561">https://twistedmatrix.com/trac/ticket/9561</a></li>
<li><a href="https://github.com/advisories/GHSA-65rm-h285-5cc5">https://github.com/advisories/GHSA-65rm-h285-5cc5</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2019-129.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/twisted/PYSEC-2019-129.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PLTZDMFBNFSJMBXYJNGJHENJA4H2TSMZ">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PLTZDMFBNFSJMBXYJNGJHENJA4H2TSMZ</a></li>
<li><a href="https://usn.ubuntu.com/4308-1">https://usn.ubuntu.com/4308-1</a></li>
<li><a href="https://usn.ubuntu.com/4308-2">https://usn.ubuntu.com/4308-2</a></li>
<li><a href="https://www.oracle.com/security-alerts/cpuapr2020.html">https://www.oracle.com/security-alerts/cpuapr2020.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00013.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00013.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00028.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00028.html</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-65rm-h285-5cc5</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2019-08-16T14:02:35.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tornado] Open redirect in Tornado]]></title>
        <id>https://github.com/advisories/GHSA-hj3f-6gcp-jg8j</id>
        <link href="https://github.com/advisories/GHSA-hj3f-6gcp-jg8j"/>
        <updated>2024-11-18T22:08:09.000Z</updated>
        <content type="html"><![CDATA[<p>Open redirect vulnerability in Tornado versions 6.3.1 and earlier allows a remote unauthenticated attacker to redirect a user to an arbitrary web site and conduct a phishing attack by having user access a specially crafted URL.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2023-28370">https://nvd.nist.gov/vuln/detail/CVE-2023-28370</a></li>
<li><a href="https://github.com/tornadoweb/tornado/releases/tag/v6.3.2">https://github.com/tornadoweb/tornado/releases/tag/v6.3.2</a></li>
<li><a href="https://github.com/tornadoweb/tornado/commit/32ad07c54e607839273b4e1819c347f5c8976b2f">https://github.com/tornadoweb/tornado/commit/32ad07c54e607839273b4e1819c347f5c8976b2f</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tornado/PYSEC-2023-75.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tornado/PYSEC-2023-75.yaml</a></li>
<li><a href="https://jvn.jp/en/jp/JVN45127776">https://jvn.jp/en/jp/JVN45127776</a></li>
<li><a href="https://github.com/advisories/GHSA-hj3f-6gcp-jg8j">https://github.com/advisories/GHSA-hj3f-6gcp-jg8j</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-hj3f-6gcp-jg8j</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2023-05-25T12:30:16.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tlslite] tlslite remote denial of service vulnerability]]></title>
        <id>https://github.com/advisories/GHSA-4749-p7rx-8jjj</id>
        <link href="https://github.com/advisories/GHSA-4749-p7rx-8jjj"/>
        <updated>2024-11-18T21:36:09.000Z</updated>
        <content type="html"><![CDATA[<p>The tlslite library before 0.4.9 for Python allows remote attackers to trigger a denial of service (runtime exception and process crash).</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2015-3220">https://nvd.nist.gov/vuln/detail/CVE-2015-3220</a></li>
<li><a href="https://github.com/trevp/tlslite/commit/aca8d4f898b436ff6754e1a9ab96cae976c8a853">https://github.com/trevp/tlslite/commit/aca8d4f898b436ff6754e1a9ab96cae976c8a853</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=1254215">https://bugzilla.redhat.com/show_bug.cgi?id=1254215</a></li>
<li><a href="https://groups.google.com/forum/#!topic/tlslite-dev/MoWE7B0A4iU">https://groups.google.com/forum/#!topic/tlslite-dev/MoWE7B0A4iU</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tlslite/PYSEC-2017-96.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tlslite/PYSEC-2017-96.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-4749-p7rx-8jjj">https://github.com/advisories/GHSA-4749-p7rx-8jjj</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-4749-p7rx-8jjj</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-14T03:07:28.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[aiohttp] aiohttp allows request smuggling due to incorrect parsing of chunk extensions]]></title>
        <id>https://github.com/advisories/GHSA-8495-4g3g-x7pr</id>
        <link href="https://github.com/advisories/GHSA-8495-4g3g-x7pr"/>
        <updated>2024-11-18T22:46:43.000Z</updated>
        <content type="html"><![CDATA[<h3 id="summary">Summary</h3>
<p>The Python parser parses newlines in chunk extensions incorrectly which can lead to request smuggling vulnerabilities under certain conditions.</p>
<h3 id="impact">Impact</h3>
<p>If a pure Python version of aiohttp is installed (i.e. without the usual C extensions) or <code>AIOHTTP_NO_EXTENSIONS</code> is enabled, then an attacker may be able to execute a request smuggling attack to bypass certain firewalls or proxy protections.</p>
<hr>
<p>Patch: <a href="https://github.com/aio-libs/aiohttp/commit/259edc369075de63e6f3a4eaade058c62af0df71">https://github.com/aio-libs/aiohttp/commit/259edc369075de63e6f3a4eaade058c62af0df71</a></p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/aio-libs/aiohttp/security/advisories/GHSA-8495-4g3g-x7pr">https://github.com/aio-libs/aiohttp/security/advisories/GHSA-8495-4g3g-x7pr</a></li>
<li><a href="https://github.com/aio-libs/aiohttp/commit/259edc369075de63e6f3a4eaade058c62af0df71">https://github.com/aio-libs/aiohttp/commit/259edc369075de63e6f3a4eaade058c62af0df71</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2024-52304">https://nvd.nist.gov/vuln/detail/CVE-2024-52304</a></li>
<li><a href="https://github.com/advisories/GHSA-8495-4g3g-x7pr">https://github.com/advisories/GHSA-8495-4g3g-x7pr</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-8495-4g3g-x7pr</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2024-11-18T21:02:32.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[aiohttp] aiohttp has a memory leak when middleware is enabled when requesting a resource with a non-allowed method]]></title>
        <id>https://github.com/advisories/GHSA-27mf-ghqm-j3j8</id>
        <link href="https://github.com/advisories/GHSA-27mf-ghqm-j3j8"/>
        <updated>2024-11-18T22:45:48.000Z</updated>
        <content type="html"><![CDATA[<h3 id="summary">Summary</h3>
<p>A memory leak can occur when a request produces a <code>MatchInfoError</code>. This was caused by adding an entry to a cache on each request, due to the building of each <code>MatchInfoError</code> producing a unique cache entry.</p>
<h3 id="impact">Impact</h3>
<p>If the user is making use of any middlewares with <code>aiohttp.web</code> then it is advisable to upgrade immediately.</p>
<p>An attacker may be able to exhaust the memory resources of a server by sending a substantial number (100,000s to millions) of such requests.</p>
<hr>
<p>Patch: <a href="https://github.com/aio-libs/aiohttp/commit/bc15db61615079d1b6327ba42c682f758fa96936">https://github.com/aio-libs/aiohttp/commit/bc15db61615079d1b6327ba42c682f758fa96936</a></p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/aio-libs/aiohttp/security/advisories/GHSA-27mf-ghqm-j3j8">https://github.com/aio-libs/aiohttp/security/advisories/GHSA-27mf-ghqm-j3j8</a></li>
<li><a href="https://github.com/aio-libs/aiohttp/commit/bc15db61615079d1b6327ba42c682f758fa96936">https://github.com/aio-libs/aiohttp/commit/bc15db61615079d1b6327ba42c682f758fa96936</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2024-52303">https://nvd.nist.gov/vuln/detail/CVE-2024-52303</a></li>
<li><a href="https://github.com/advisories/GHSA-27mf-ghqm-j3j8">https://github.com/advisories/GHSA-27mf-ghqm-j3j8</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-27mf-ghqm-j3j8</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2024-11-18T21:02:17.000Z</published>
    </entry>
</feed>